File: scalechain-util/src/main/scala/io/scalechain/util/StackUtil.scala
package io.scalechain.util

import java.io.{PrintWriter, StringWriter}

/**
  * Created by kangmo on 3/22/16.
  */
object StackUtil {
  def getStackTrace(e : Throwable) : String = {
    val stringWriter = new StringWriter
    val writer = new PrintWriter(stringWriter)
    e.printStackTrace(writer)
    stringWriter.toString
  }

  def getCurrentStack() : String = {
    val stacktrace = Thread.currentThread().getStackTrace()
    stacktrace.mkString("\n")
  }
}

File: scalechain-util/src/main/kotlin/io/scalechain/util/StackUtil.kt
package io.scalechain.util

import java.io.PrintWriter
import java.io.StringWriter


/**
 * Created by kangmo on 3/22/16.
 */
object StackUtil {
    @JvmStatic
    fun getStackTrace(e : Throwable) : String {
        val stringWriter = StringWriter()
        val writer = PrintWriter(stringWriter)
        e.printStackTrace(writer)
        return stringWriter.toString()
    }

    @JvmStatic
    fun getCurrentStack() : String {
        val stacktrace = Thread.currentThread().getStackTrace()
        return stacktrace.joinToString("\n")
    }
}
